src/index.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  getInput,
+  getMultilineInput,
+  info,
+  setFailed,
+  endGroup,
+  startGroup,
+} from "@actions/core";
+import { spawnSync } from "node:child_process";
+import { existsSync } from "node:fs";
+import * as path from "node:path";
+import * as util from "util";
+const exec = util.promisify(require("child_process").exec);
+
+const DEFAULT_WRANGLER_VERSION = "^3.0.0";
+
+/**
+ * A configuration object that contains all the inputs & immutable state for the action.
+ */
+const config = {
+  WRANGLER_VERSION:
+    getInput("wranglerVersion") !== ""
+      ? getInput("wranglerVersion")
+      : DEFAULT_WRANGLER_VERSION,
+  secrets: getMultilineInput("secrets"),
+  workingDirectory: checkWorkingDirectory(getInput("workingDirectory")),
+  CLOUDFLARE_API_TOKEN: getInput("apiToken"),
+  CLOUDFLARE_ACCOUNT_ID: getInput("accountId"),
+  ENVIRONMENT: getInput("environment"),
+  VARS: getMultilineInput("vars"),
+  COMMANDS: getMultilineInput("commands"),
+} as const;
+
+async function main() {
+  installWrangler();
+  authenticationSetup();
+  await execCommands(getMultilineInput("preCommands"));
+  await uploadSecrets();
+  await wranglerCommands();
+  await execCommands(getMultilineInput("postCommands"));
+}
+
+function checkWorkingDirectory(workingDirectory = "") {
+  try {
+    const normalizedPath = path.normalize(workingDirectory);
+    if (existsSync(normalizedPath)) {
+      return normalizedPath;
+    } else {
+      setFailed(`🚨 Directory ${workingDirectory} does not exist.`);
+    }
+  } catch (error) {
+    setFailed(
+      `🚨 While checking/creating directory ${workingDirectory} received ${error}`
+    );
+  }
+}
+
+function installWrangler() {
+  if (config["WRANGLER_VERSION"].startsWith("1")) {
+    setFailed(`🚨 Wrangler v1 is no longer supported by this action. Please use major versions 2 or 3`);
+  }
+  startGroup("📥 Installing Wrangler");
+  const command = `npx install wrangler@${config["WRANGLER_VERSION"]}`;
```

Comment:
Should this be `npm`?

Reply:
How does this work if a user has specified a wrangler version in their `package.json`, but not in this action—should we try and detect that somehow?

Reply:
I dont think the previous Action did that, sounds like a great additional feature.